### PR TITLE
fix(core): unable to open block style select when in nested PTEs in fullscreen

### PR DESCRIPTION
**Pull request by RitaDias**
### Description

Fix issue where if you had nested PTEs you would not be able to open the block style dropdown

Before

https://github.com/sanity-io/sanity/assets/6951139/21ef9ef6-6de0-4ba5-b593-ea579fbf3fd5

### What to review

The code makes sense, the tests make sense.
Should we add some more tests? Thought about component but since the dropdown itself existed in the UI just had the "hidden" property I considered that the e2e was the best approach. Let me know if that is not correct

### Testing
Tests are written
You can also manually test on the All Bells and Whistles on the last PTE: content

### Notes for release

Fixes issue where nested fullscreen PTEs were unable to open the block style dropdown

### CODE REVIEW

**Comment by RitaDias** [comment]
For context: There seems to be some current issues on the `Check if the React Compiler are able to optimize everything / lint` that is currently unrelated to this PR

**Review by binoy14 — APPROVED** [review]
working as expected, thanks!

File: test/e2e/tests/pte/Toolbar.spec.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import {expect, type Locator} from '@playwright/test'
+import {test} from '@sanity/test'
+
+test.describe('Portable Text Input - Open Block Style Select', () => {
+  let pteInput: Locator
+
+  test.beforeEach(async ({page, createDraftDocument}) => {
```
[the remark this answers is not in the document]
Comment (RitaDias) [reply test/e2e/tests/pte/Toolbar.spec.ts:7]: I also didn't til this 😄

**Comment by RitaDias** [comment]
Noticed that one of the tests wasn't waiting for the nested full screen, hence the update

**Pull request merged**